pkg/services/ngalert/schedule.go

```diff
@@ -0,0 +1,215 @@
+package ngalert
+
+import (
+	"context"
+	"sync"
+	"time"
+)
+
+func (ng *AlertNG) definitionRoutine(definitionID int64, evalCh <-chan *evalContext) {
```

Comment:
I would prefer of the Scheduler was a separate component/service and not part of the main AlertNG service.